### ISSUE TITLE
[Compiler Bug]:  babel-plugin-react-compiler@0.0.0-experimental-7779988-20241007

### ISSUE THREAD

**Issue by kmani-fullstack** [facebook/react#31150]
### What kind of issue is this?

- [ ] React Compiler core (the JS output is incorrect, or your app works incorrectly after optimization)
- [X] babel-plugin-react-compiler (build issue installing or using the Babel plugin)
- [ ] eslint-plugin-react-compiler (build issue installing or using the eslint plugin)
- [ ] react-compiler-healthcheck (build issue installing or using the healthcheck script)

### Link to repro

https://www.npmjs.com/package/babel-plugin-react-compiler

### Repro steps

[2024-10-08T04_56_44_562Z-debug-0.log](https://github.com/user-attachments/files/17288171/2024-10-08T04_56_44_562Z-debug-0.log)
[2024-10-08T04_56_44_562Z-debug-0.log](https://github.com/user-attachments/files/17288179/2024-10-08T04_56_44_562Z-debug-0.log)

Unable to create new expo project. Error when npm install in new project.

MAC Mini Apple M2, macOS 14.7

Fix:
Latest version  babel-plugin-react-compiler@0.0.0-experimental-7779988-20241007. contains problem. I replaced my old react expo project that works with 0.0.0-experimental-734b737-20241003

### How often does this bug happen?

Every time

### What version of React are you using?

react - 18.2.0, react-native - 0.74.5

**Comment by ppaann** [comment]
Same here! complain about the same missing file during installation:
`error sh: ./scripts/link-react-compiler-runtime.sh: No such file or directory`

info run babel-plugin-react-compiler@0.0.0-experimental-7779988-20241007 postinstall node_modules/babel-plugin-react-compiler ./scripts/link-react-compiler-runtime.sh
7343 info run babel-plugin-react-compiler@0.0.0-experimental-7779988-20241007 postinstall { code: 127, signal: null }
7344 timing reify:rollback:createSparse Completed in 8072ms
7345 timing reify:rollback:retireShallow Completed in 0ms
7346 timing command:install Completed in 46318ms
7347 verbose stack Error: command failed
7347 verbose stack     at ChildProcess.<anonymous> (/Users/Pan/.nvm/versions/node/v21.6.2/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
7347 verbose stack     at ChildProcess.emit (node:events:519:28)
7347 verbose stack     at maybeClose (node:internal/child_process:1105:16)
7347 verbose stack     at Socket.<anonymous> (node:internal/child_process:457:11)
7347 verbose stack     at Socket.emit (node:events:519:28)
7347 verbose stack     at Pipe.<anonymous> (node:net:337:12)
7348 verbose pkgid babel-plugin-react-compiler@0.0.0-experimental-7779988-20241007
7349 verbose cwd /Users/Pan/Projects/ShoppingList/my-app
7350 verbose Darwin 22.6.0
7351 verbose node v21.6.2
7352 verbose npm  v10.2.4
7353 error code 127
7354 error path /Users/Pan/Projects/ShoppingList/my-app/node_modules/babel-plugin-react-compiler
7355 error command failed
7356 error command sh -c ./scripts/link-react-compiler-runtime.sh
7357 error sh: ./scripts/link-react-compiler-runtime.sh: No such file or directory
7358 verbose exit 127
7359 timing npm Completed in 46986ms
7360 verbose unfinished npm timer reify 1728367592806
7361 verbose unfinished npm timer reify:build 1728367630926
7362 verbose unfinished npm timer build 1728367630928
7363 verbose unfinished npm timer build:deps 1728367630929
7364 verbose unfinished npm timer build:run:postinstall 1728367631016
7365 verbose unfinished npm timer build:run:postinstall:node_modules/babel-plugin-react-compiler 1728367631017
7366 verbose code 127

**Comment by cozyClassic** [comment]
Same here

**Comment by BoChingLee** [comment]
Same issue

**Comment by Macarthurval** [comment]
Same problem

**Comment by Jl115** [comment]
Same

**Comment by compojoom** [comment]
I'm getting the same bug. 
This causes `npx create-expo-app@latest` to fail installing, which means that every user who tries to start developing with React-Native today will have a hard time...

**Comment by Jl115** [comment]
> I'm getting the same bug. This causes `npx create-expo-app@latest` to fail installing, which means that every user who tries to start developing with React-Native today will have a hard time...

and also when you update...

**Comment by compojoom** [comment]
I see that a fix has been commited to the repo: https://github.com/facebook/react/commit/ed966dac4a025fd37580e8197e5b271044ffbd9f
I guess that a new release will follow soon.

**Comment by Jl115** [comment]
Temp fix is in package.json: 

"overrides": {
    "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003"
  },

**Comment by dinoyee** [comment]
Same issue

**Comment by mimatik** [comment]
+ 1

**Comment by bzyfuzy** [comment]
same :(

**Comment by sharif2008** [comment]
also faced the same issue. Temp fix -

```
"resolutions": {
 "**/babel-plugin-react-compiler": "0.0.0-experimental-ca8e0be-20240916"
},
```

**Comment by r0dz** [comment]
same here

**Comment by erdemozkok** [comment]
same error

**Comment by poteto** [comment]
This has been fixed and released in [0.0.0-experimental-58c2b1c-20241007](https://www.npmjs.com/package/babel-plugin-react-compiler/v/0.0.0-experimental-58c2b1c-20241007), sorry about that!

Please reopen the issue if the issue persists.
